addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1028,8 +1039,19 @@ options.registry.layout_column = options.Class.extend({
      * @see this.selectClass for parameters
      */
     selectCount: async function (previewMode, widgetValue, params) {
+        let $row = this.$('> .row:not(.o_we_zero_cols_content)');
+        if (!$row.length) {
+            this.$target.contents().wrapAll($('<div class="row"><div class="col-lg-12"/></div>'));
+            $row = this.$('> .row');
+        }
+
         const nbColumns = parseInt(widgetValue);
-        await this._updateColumnCount(nbColumns - this.$target.children().length);
+        await this._updateColumnCount($row, (nbColumns || 1) - $row.children().length);
+        if (nbColumns === 0) {
+            $row.contents().unwrap()
+                .contents().unwrap()
+                .addClass('o_we_zero_cols_content');
```

Comment:
Can't you reverse the logic and have a class on row to consider instead of adding a class on every content a container without row can have?

Reply:
This is for the hypothetical case where we allow unwrapping a row-in-col snippet (like the features grid snippet), but since we don't allow that at the moment it might not be needed at all. And if we do need it, we can indeed add a class like `o_we_no_unwrap` on the rows of the snippet in question.